userspace/libscap/examples/01-open/scap_open.c

```diff
@@ -20,42 +20,328 @@ limitations under the License.
+
+/* SCAP SOURCES */
 #define KMOD_OPTION "--kmod"
 #define BPF_OPTION "--bpf"
 #define MODERN_BPF_OPTION "--modern_bpf"
 #define SCAP_FILE_OPTION "--scap_file"
-#define SIMPLE_CONSUMER_OPTION "--simple_consumer"
+
+/* CONFIGURATIONS */
+#define TP_OPTION "--tp"
+#define PPM_SC_OPTION "--ppm_sc"
 #define NUM_EVENTS_OPTION "--num_events"
 #define EVENT_TYPE_OPTION "--evt_type"
+
+/* PRINT */
 #define VALIDATION_OPTION "--validate_syscalls"
 #define PRINT_SYSCALLS_OPTION "--print_syscalls"
 #define PRINT_HELP_OPTION "--help"
 
-#define SYSCALL_NAME_MAX_LEN 40
-#define UNKNOWN_ENGINE "unknown"
-#define UNKNOWN_ENGINE_LEN 7
-
 extern const struct ppm_syscall_desc g_syscall_info_table[PPM_SC_MAX];
 extern const struct ppm_event_info g_event_info[PPM_EVENT_MAX];
 extern const struct syscall_evt_pair g_syscall_table[SYSCALL_TABLE_SIZE];
 extern const enum ppm_syscall_code g_syscall_code_routing_table[SYSCALL_TABLE_SIZE];
 
 /* Engine params */
-struct scap_bpf_engine_params bpf_params;
-struct scap_savefile_engine_params savefile_params;
+struct scap_bpf_engine_params bpf_params = {0};
+struct scap_savefile_engine_params savefile_params = {0};
 
 /* Configuration variables set through CLI. */
 uint64_t num_events = UINT64_MAX; /* max number of events to catch. */
-bool simple_consumer = false;	  /* kernel simple consumer mode. */
 int evt_type = -1;		  /* event type to print. */
+bool ppm_sc_is_set = 0;
+bool tp_is_set = 0;
 
 /* Generic global variables. */
-scap_open_args oargs = {.engine_name = UNKNOWN_ENGINE}; /* scap oargs used in `scap_open`. */
-uint64_t g_nevts = 0;				   /* total number of events captured. */
-scap_t* g_h = NULL;				   /* global scap handler. */
-uint16_t* lens16 = NULL;			   /* pointer used to print the length of event params. */
-char* valptr = NULL;				   /* pointer used to print the value of event params. */
+scap_open_args oargs = {.engine_name = UNKNOWN_ENGINE};			    /* scap oargs used in `scap_open`. */
+uint64_t g_nevts = 0;							    /* total number of events captured. */
+scap_t* g_h = NULL;							    /* global scap handler. */
+uint16_t* lens16 = NULL;						    /* pointer used to print the length of event params. */
+char* valptr = NULL; /* pointer used to print the value of event params. */ /* pointer used to print the value of event params. */
+struct timeval tval_start, tval_end, tval_result;
+
+/*=============================== PRINT SUPPORTED SYSCALLS ===========================*/
+
+void print_sorted_syscalls(char string_vector[SYSCALL_TABLE_SIZE][SYSCALL_NAME_MAX_LEN], int dim)
+{
+	char temp[SYSCALL_NAME_MAX_LEN];
+
+	/* storing strings in the lexicographical order */
+	for(int i = 0; i < dim; ++i)
+	{
+		for(int j = i + 1; j < dim; ++j)
+		{
+			/* swapping strings if they are not in the lexicographical order */
+			if(strcmp(string_vector[i], string_vector[j]) > 0)
+			{
+				strcpy(temp, string_vector[i]);
+				strcpy(string_vector[i], string_vector[j]);
+				strcpy(string_vector[j], temp);
+			}
```

Comment:
I feel like you could probably use an auxiliar array with pointers to the names and prevent both copying syscalls around and modifying the input vector, but seeing as this is an example binary we don't need to bother with it.

Reply:
Yeah agree, no need to bother with this kind of improvements IMHO!